ethereum/blockwatch/block_watcher.go

```diff
@@ -586,38 +578,17 @@ type blockRange struct {
 // blocks' logs twice.
 func (w *Watcher) getSubBlockRanges(from, to, rangeSize int) []*blockRange {
 	chunks := []*blockRange{}
-	numBlocksLeft := to - from
-	if numBlocksLeft < rangeSize {
+	for from+rangeSize-1 < to {
```

Comment:
How do you feel about moving `from += rangeSize` to the post condition of the for loop? That way it guarantees we won't forget to do it.

```suggestion
	for ; from+rangeSize-1 < to; from += rangeSize {
```

Reply:
Works for me!